components/uniset/src/ule.rs

```diff
@@ -0,0 +1,54 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::enum_props::{GeneralSubcategory, Script};
+use core::convert::TryFrom;
+use num_enum::TryFromPrimitiveError;
+use zerovec::ule::{AsULE, PlainOldULE, ULE};
+
+#[repr(transparent)]
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct GeneralSubcategoryULE(u8);
+
+impl AsULE for GeneralSubcategory {
+    type ULE = GeneralSubcategoryULE;
+
+    #[inline]
+    fn as_unaligned(&self) -> Self::ULE {
+        let u = *self as u8;
+        GeneralSubcategoryULE(u)
+    }
+
+    #[inline]
+    fn from_unaligned(unaligned: &Self::ULE) -> Self {
+        // Safe because the contents of GeneralSubcategoryULE are required to be valid.
+        unsafe { Self::from_unchecked(unaligned.0) }
+    }
+}
+
+unsafe impl ULE for GeneralSubcategoryULE {
```

Comment:
issue: Please add a comment explaining why this impl satisfied the invariants of ULE

(yes, I know that the ZeroVec impls don't consistently do this, but I'm hoping to move towards always documenting this

Reply:
+1 and see https://github.com/unicode-org/icu4x/pull/1121 for an updated safety checklist for ULE